tests/ecosystem/deployment/test_ocs_basic_install.py

```diff
@@ -177,6 +202,29 @@ def test_deployment(self):
         log.info(f"Waiting {wait_time} seconds...")
         time.sleep(wait_time)
 
+        # Create MDS pods for CephFileSystem
+        self.fs_data = defaults.CEPHFILESYSTEM_DICT.copy()
```

Comment:
just copy is not enough cause it's just shallow copy, but if it contains the nested dictionaries it's pointing still to origin dictionaries which is very fragile and you will change the values in origin CEPHFILESYSTEM_DICT.  See this one output from ipython to understand this:
```python
In [8]: myd = {'metadata': {'namespace': "default_namespace", 'name': 'test'}}

In [9]: myd_copy = myd.copy()

In [10]: myd
Out[10]: {'metadata': {'namespace': 'default_namespace', 'name': 'test'}}

In [11]: myd_copy
Out[11]: {'metadata': {'namespace': 'default_namespace', 'name': 'test'}}

In [12]:

In [12]: myd_copy['metadata']['namespace'] = "NewNamespace!!!"

In [13]: myd_copy
Out[13]: {'metadata': {'namespace': 'NewNamespace!!!', 'name': 'test'}}

In [14]: myd
Out[14]: {'metadata': {'namespace': 'NewNamespace!!!', 'name': 'test'}}
```

so please use rather [deepcopy](https://docs.python.org/2/library/copy.html#copy.deepcopy)

Reply:
+1 The root of the problem seems to be in the fact that we use nested structures in defaults module, which is prone to such errors. It would be better to have a function or a fixture which tests will use the get the private copy of the structure (so that test authors can't run into this issue by mistake).

Reply:
good catch!

Reply:
@petr-balogh @mbukatov @vasukulkarni thanks for sharing the insights on copy().

Should I then use following:
self.fs_data = defaults.CEPHFILESYSTEM_DICT.deepcopy() 
I shall try my install with this and get back with results.

T

Reply:
Resent the PR . 
Please review.